File: poc/bank-simulator/src/main/java/eu/accesa/blinkpay/bank/dto/OfflineSyncRequest.java
package eu.accesa.blinkpay.bank.dto;

import java.util.List;

/**
 * POST /bank/wallet/{walletId}/sync-offline-transactions
 *
 * Batch of offline NFC P2P transfers to reconcile with the bank.
 * Each transaction carries a unique ID for idempotent processing.
 */
public record OfflineSyncRequest(List<OfflineTransactionEntry> transactions) {

    public record OfflineTransactionEntry(
            String transactionId,       // UUID generated on mobile at transfer time
            String counterpartyIban,
            java.math.BigDecimal amount,
            Direction direction,        // SEND or RECEIVE
            java.time.Instant timestamp
    ) {}

    public enum Direction { SEND, RECEIVE }
}

File: poc/mobile-client/app/src/main/java/eu/accesa/blinkpay/data/api/dto/OfflineSyncRequest.kt
package eu.accesa.blinkpay.data.api.dto

import java.math.BigDecimal

data class OfflineSyncRequest(
    val transactions: List<OfflineTransactionEntry>,
)

data class OfflineTransactionEntry(
    val transactionId: String,
    val counterpartyIban: String,
    val amount: BigDecimal,
    val direction: String,   // "SEND" or "RECEIVE"
    val timestamp: String,   // ISO-8601 instant
)
